src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -599,6 +649,43 @@ def __init__(self, children=None, parent=None, grainsize=None,
                 "numtasks clauses specified.")
         super(OMPTaskloopDirective, self).__init__(children=children,
                                                    parent=parent)
+        if self._grainsize is not None:
+            child = [Literal(f"{grainsize}", INTEGER_TYPE)]
+            self._children.append(OMPGrainsizeClause(children=child))
+        if self._num_tasks is not None:
+            child = [Literal(f"{num_tasks}", INTEGER_TYPE)]
+            self._children.append(OMPNumTasksClause(children=child))
+        if self._nogroup:
+            self._children.append(OMPNogroupClause())
+
+    @staticmethod
+    def _validate_child(position, child):
+        '''
+         Decides whether a given child and position are valid for this node.
+         The rules are:
+         1. Child 0 must always be a Schedule.
+         2. Child 1 may be either a OMPGrainsizeClause or OMPNumTasksClause, \
+            or if neither of those clauses are present, it may be a \
+            OMPNogroupClause.
+         3. Child 2 must always be a OMPNogroupClause, and can only exist if \
+            child 1 is a OMPGrainsizeClause or OMPNumTasksClause.
+
+        :param int position: the position to be validated.
+        :param child: a child to be validated.
+        :type child: :py:class:`psyclone.psyir.nodes.Node`
+
+        :return: whether the given child and position are valid for this node.
+        :rtype: bool
+
+        '''
+        if position == 0:
+            return isinstance(child, Schedule)
+        if position == 1:
+            return isinstance(child, (OMPGrainsizeClause, OMPNumTasksClause,
+                                      OMPNogroupClause))
+        if position == 2:
+            return (isinstance(child, OMPNogroupClause))
```

Comment:
Pylint complaints about the unneeded outer parenthesis

Reply:
Fixed